libyul/optimiser/ASTWalker.h

```diff
@@ -102,4 +102,32 @@ class ASTModifier
 	}
 };
 
+template <
+	typename Node,
+	typename Visitor,
+	typename Base = std::conditional_t<std::is_const_v<Node>, ASTWalker, ASTModifier>
+>
+struct ForEach: Base
+{
+	template<typename Callable>
+	ForEach(Callable&& _visitor): visitor(std::forward<Callable>(_visitor)) {}
```

Comment:
Why do we need the constructor?

Reply:
Since the base classes have virtual functions, aggregate initialization won't work.. or sth like that :-)...
I templated it to be sure that the forwarding works correctly.

But in general, the class itself is not meant for general use - I might even put it into ``namespace detail`` to clarify that ``forEach`` is the proper interface and this is merely a technical helper...

Reply:
moved it into detail.